LFS-426 / LFS-428: Patient chart UI

anusha-dey opened the pull request:
Issue Link: https://phenotips.atlassian.net/browse/LFS-428

To test create forms for a subject and forms for its child SubjectTypes, all the forms/form previews should appear on the subject page

** this branch has changes from LFS-426_LFS-436 so that the SubjectType node could be used. The changes for this component are just in QuestionnaireStyle.jsx and Subject.jsx

anusha-dey committed:
pass form id from livetable to subject

anusha-dey committed:
use form id to display form questions; wip

anusha-dey committed:
fetch data from all related forms, working on rendering needed data in table

anusha-dey committed:
fetch and display data in subjects file

anusha-dey committed:
display preview of form. todo: fix radiotherapy data

anusha-dey committed:
start implementing cards

anusha-dey committed:
fetch data to fill cards

anusha-dey committed:
render form data into card, todo: refactor FormData

anusha-dey committed:
refactored, added new recursive component to handle related subjects, start adding styling

anusha-dey committed:
add styling to cards

anusha-dey committed:
wip: handle displaying sections

anusha-dey committed:
handle displaying sections from forms

anusha-dey committed:
only display 4 form entries

anusha-dey committed:
use recursive function to display subjects

anusha-dey committed:
fix style

anusha-dey committed:
Merge branch 'LFS-426_LFS-436' of https://github.com/ccmbioinfo/lfs into LFS-428

anusha-dey committed:
Merge branch 'LFS-426_LFS-436' of https://github.com/ccmbioinfo/lfs into LFS-428

anusha-dey committed:
Merge branch 'LFS-426_LFS-436' of https://github.com/ccmbioinfo/lfs into LFS-428

anusha-dey committed:
add subjecttype, display related subjects (wip)

anusha-dey committed:
display all related subjecttypes, add comments

anusha-dey committed:
create button to add related subjects

anusha-dey committed:
add comments, fix loading issue, fix button for creating new forms

anusha-dey committed:
fix displaying form data

anusha-dey committed:
whitespace fix

anusha-dey committed:
remove unnecessary changes

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/QuestionnaireStyle.jsx:
```diff
@@ -120,6 +120,27 @@ const questionnaireStyle = theme => ({
+        padding: "0 !important"
+    },
+    subjectHeader: {
+        paddingBottom: "12px !important",
```
Use theme.spacing instead

acrowthe replied:
subjectHeader was unused: removed

anusha-dey committed:
Merge branch 'dev' of https://github.com/ccmbioinfo/lfs into LFS-428

anusha-dey committed:
pass current subject data to newformdialog

anusha-dey committed:
create form if requiredsubjecttype of form is the current subjec type

anusha-dey committed:
handle creating form for all subject types from subject page

anusha-dey committed:
fix merge conflicts

anusha-dey committed:
wip, todo:only display related questionnaires/subjects

sashaandjic commented:
New subject bttn does not work
![image](https://user-images.githubusercontent.com/23243244/84947702-a19b4500-b0b8-11ea-9c75-9e5809dc8a8a.png)

sashaandjic commented:
Patient chart appears to be as per specifications. Should there be a limit for certain forms ( e.g. Can a subject have more than one demographic forms?). Also, if there are multiple forms of the same type, hpw should they be displayed? Currently forms are displayed in chronological order.
![image](https://user-images.githubusercontent.com/23243244/84947907-0787cc80-b0b9-11ea-9c97-110c5b926280.png)

anusha-dey committed:
wip, only show related subjects

sashaandjic commented:
Not sure if this was introduced in this PR, but when creating Tumors the available subjects are not displayed
![image](https://user-images.githubusercontent.com/23243244/84959300-12e4f300-b0cd-11ea-9723-f0237d89c2f0.png)

sdumitriu reviewed (dismissed):
When a questionnaire supports more than one type of subject, prompt the user to select a subject.

- [x] Questionnaire requires one subject type, and the current subject has that type, it is used for the new form (this is currently correct)
- [x] Questionnaire requires several subject types, and the current subject is one of those types, it should still prompt to choose the subject (this is currently wrong)
- [x] Current subject has a type that is not supported by the selected questionnaire, then it should prompt for the subject (this is currently correct)

Example: the CNV questionnaire can be filled in for either a patient, a tumor, or a tumor region. When viewing a patient and adding a CNV form, the user should be able to choose whether it applies to the current patient, one of its tumors, one of its tumor's regions, or a new patient.

anusha-dey committed:
only display related subjects (complete)

anusha-dey committed:
handle if there is more than 1 requiredSubjectTypes

qrst07 commented:
@sdumitriu the dialog on the subject page should now prompt to choose the subject if the questionnaire requires several subjecttypes

anusha-dey committed:
remove some comments

marta- commented:
> Should there be a limit for certain forms ( e.g. Can a subject have more than one demographic forms?)

@sashaandjic I created https://phenotips.atlassian.net/browse/LFS-473

sashaandjic commented:
I mistakenly created Chemotherapy form for patient Mary Green. When I opened the form and clicked on Mary Green to modify the patient the dialog showed me patients and tumours to choose from. It should only offered me the type of patient, since I clicked on the patient.
![image](https://user-images.githubusercontent.com/23243244/85174020-7fcdc980-b242-11ea-8e0f-993e524a5cd5.png)

sashaandjic commented:
I believe all of my complaints, except for broken NEW subject button, were addressed in LFS-458 https://github.com/ccmbioinfo/lfs/pull/226.

veronikaslc commented:
@qrst07 we can merge after conflicts are resolved or need re-testing?

anusha-dey committed:
Merge branch 'dev' of https://github.com/ccmbioinfo/lfs into LFS-428

anusha-dey committed:
fix conflict, todo: automatically assign parent

anusha-dey committed:
create form, set currentSubject as parent

anusha-dey committed:
wip: only display related subjects

anusha-dey committed:
only display related subjects in dialog

anusha-dey committed:
Merge branch 'dev' of https://github.com/ccmbioinfo/lfs into LFS-428

sashaandjic reviewed (approved):
tested

anusha-dey committed:
Merge branch 'dev' of https://github.com/ccmbioinfo/lfs into LFS-428

sashaandjic commented:
Regression found after the last change. Clicking on a subject link gives me a blank screen.

acrowthe committed:
Merge branch 'dev' into LFS-428

acrowthe committed:
LFS-426/LFS-428: Patient chart UI
Fix merge regression

acrowthe committed:
LFS-426/LFS-428: Patient chart UI
Slight adjustments
- Comment fixes and removal
- Refactor ternary
- Adjust FAB location

acrowthe commented:
Looks good to me

sashaandjic reviewed (approved):
re-tested; all good; Somatic variants data forms are added
please note that this branch is missing recently done LFS-501-fix

acrowthe committed:
LFS-426/LFS-428: Patient chart UI
Remove unused css class

sashaandjic commented:
Problem that Sergiu found is logged as LFS-573 - Values displayed in patient charts are the raw stored values, not the user-friendly ones
![LFS-573](https://user-images.githubusercontent.com/23243244/93838181-c292a980-fc56-11ea-84b9-319c9ae7da17.png)
